rule.go

```diff
@@ -458,6 +458,17 @@ func (r *Rule) executeTransformations(value string) (string, []error) {
 	return value, errs
 }
 
+func (r *Rule) Clone() *Rule {
+	clone := &Rule{
+		RuleMetadata:    r.RuleMetadata,
```

Comment:
Metadata is a copy, but we must take care of the internal pointers, like the arrays